### PR TITLE
Fix performance bug

### PR DESCRIPTION
**Pull request by imeckler**
I accidentally committed some debug lines which were computing completely unused polynomial commitments inside the opening function. This removes that and also optimizes the code for combining polynomials to an improved version. In testing this improved prover performance by about 15-20% :sweat_smile:

### CODE REVIEW

File: utils/src/dense_polynomial.rs [hunk 1]
```diff
@@ -28,9 +29,9 @@ pub trait ExtendedDensePolynomial<F: Field> {
 impl<F: Field> ExtendedDensePolynomial<F> for DensePolynomial<F> {
     fn scale(&self, elm: F) -> Self {
         let mut result = self.clone();
-        for coeff in &mut result.coeffs {
+        result.coeffs.par_iter_mut().for_each(|coeff| {
```
Comment (mimoo) [inline utils/src/dense_polynomial.rs]: I think `for coeff in result.coeffs.par_iter_mut() {` should work no?

File: dlog/commitment/src/commitment.rs [hunk 2]
```diff
@@ -660,62 +665,50 @@ where
...
         g.extend(vec![G::zero(); padding]);
 
-        let mut combined_polynomial_terms = vec![];
-
-        // scale the polynoms in accumulator shifted, if bounded, to the end of SRS
         let (p, blinding_factor) = {
-            let mut p = DensePolynomial::<Fr<G>>::zero();
-
-            let mut omega = Fr::<G>::zero();
+            let start = std::time::Instant::now();
```
Comment (mimoo) [inline dlog/commitment/src/commitment.rs]: btw it feels weird to have these timers in prod code. Wouldn't it be better to extract this logic in its own function and then write a bench for it?

File: dlog/commitment/src/commitment.rs [hunk 3]
```diff
@@ -677,45 +679,44 @@ where
                 if let Some(m) = degree_bound {
                     assert!(p_i.coeffs.len() <= m + 1);
                     while j < omegas.unshifted.len() {
-                        let segment = DensePolynomial::<Fr<G>>::from_coefficients_slice(
-                            &p_i.coeffs[offset..if offset + self.g.len() > p_i.coeffs.len() {
+                        let segment = &p_i.coeffs[offset
+                            ..if offset + self.g.len() > p_i.coeffs.len() {
```
Comment (mimoo) [inline dlog/commitment/src/commitment.rs:683]: unrelated to the PR, but this inlined `if` should really be rewritten

File: dlog/commitment/src/commitment.rs [hunk 4]
```diff
@@ -724,7 +725,33 @@ where
...
+                            res.append(&mut v);
+                            DensePolynomial::from_coefficients_vec(res)
+                        }
+                    }
+                })
+                .collect();
+
+            for p in scaled {
+                res += &p;
+            }
```
Comment (mimoo) [inline dlog/commitment/src/commitment.rs:752]: could this logic of reconstructing a polynomial from its chunks be abstracted in `utils/src/polynomials.rs` or something like that? 

```rust
pub trait ChunkedPolynomial: Default {
  fn add_unshifted(&mut self, scale: F, coeffs: Vec<F>)
  fn add_shifted(&mut self, scale: F, coeffs: Vec<F>)
  fn reconstruct(self) -> DensePolynomial
}
```

**Review by mimoo — APPROVED** [review]
LGTM, left you a few comments

File: dlog/commitment/src/commitment.rs (hunk 2 above)
Comment (imeckler) [reply dlog/commitment/src/commitment.rs]: Hm I think I have deleted this in a later commit

**Comment by jspada** [comment]
nice catch and nice performance jump

File: utils/src/dense_polynomial.rs (hunk 1 above)
Comment (imeckler) [reply utils/src/dense_polynomial.rs]: Doesn't work actually, only works for `Iterator`s, not the parallel iterator trait from rayon.

File: dlog/commitment/src/commitment.rs (hunk 4 above)
Comment (imeckler) [reply dlog/commitment/src/commitment.rs:752]: Thanks, did something like this

**Pull request merged**